src/template/search-form.ts

```diff
@@ -0,0 +1,122 @@
+import { html, render } from 'lit-html'
+import { fetchData } from '../utils/fetchData'
+import { mainTemplate } from './main'
+import { urlParameterMap } from '../utils/urlParameter'
+import { relatedResultsTemplate } from './related-search'
+import { noResultsTemplate } from './no-results'
+
+export function searchForm () {
+  const currUrlParameterMap = urlParameterMap()
+
+  const onSubmit = (e: any) => {
+    e.preventDefault()
+    const params = new URLSearchParams(location.search)
+    const currUrlParameterMap = urlParameterMap()
+
+    // set params
+    params.set('scope', currUrlParameterMap.scope)
+    params.set('profile', currUrlParameterMap.profile)
+    params.set('page', '1')
+    params.set('start_rank', '1')
+    params.set('num_ranks', '10')
+    params.set('tiers', 'off')
+    params.set('collection', 'qld-gov')
+
+    // push history stack and fetch data
+    setTimeout(function () {
+      const currInputValue = (document.querySelector('.qg-site-search__component .qg-search-site__input') as HTMLInputElement).value
+      params.set('query', currInputValue)
+      history.pushState({}, '', `?${params.toString()}`)
+      fetchData(params.toString()).then(data => {
+        const { contextualNavigation, results } = data?.response?.resultPacket
+        if (results?.length) {
+          render(mainTemplate(data?.response, currUrlParameterMap), document.getElementById('qg-search-results__container') as HTMLBodyElement)
+          render(relatedResultsTemplate(contextualNavigation), document.getElementById('related-search__tags')!)
+        } else {
+          render(noResultsTemplate(), document.getElementById('qg-search-results__container')!)
+          // TODO - include related search in the main template
+          render(relatedResultsTemplate(contextualNavigation), document.getElementById('related-search__tags')!)
+        }
+      })
+    })
+  }
+
+  const suggestionVisibility = (value: boolean) => {
+    const suggestionEl = document.querySelector('.qg-site-search__component .qg-search-concierge-help') as HTMLElement
+    if (suggestionEl != null) {
+      suggestionEl.hidden = value
+    }
+  }
+
+  const onInputClick = (e: { type: string; key: string }) => {
+    if (e.type === 'keydown') {
+      if (e.key === 'Enter') {
+        suggestionVisibility(true)
+      } else {
+        suggestionVisibility(false)
+      }
+    }
+  }
+
+  return html`
+        <form action="#" role="search" class="qg-site-search__form qg-site-search__component qg-search-form" data-suggestions="https://find.search.qld.gov.au/s/suggest.json?collection=qld-gov&fmt=json%2B%2B&alpha=0.5&profile=qld" data-results-url="https://find.search.qld.gov.au/s/search.json?collection=qld-gov&profile=qld&meta_sfinder_sand=yes">
```

Comment:
should this be a template file like no-result.ts ?

Reply:
Currently, I am structuring template files in a way that the logic related to the template is at the same place to ease the process of debugging.

Template file -> Template HTML + Template Logic (JS)

'No results found' does not have any logic but the 'search form' needs logic on clicking the button etc.